### PR TITLE
Has alpha

### LINKED ISSUE #271: image.hasAlpha() Method?
Is it possible to add a method for determining if a PNG has a transparency colour channel?  

Creating alpha channels when none is required results in unnecessarily large images.  

Ideally, the following could be used:
```js
image.rgba( image.hasAlpha() );
```
This would either preserve the channel when used remove it when unecessary.  

Alternatively you could use it to determine file types:
```js
if ( image.hasAlpha() ) {
    file = "filename.png";
} else {
    file = "filename.jpg";
}
```

### PR DESCRIPTION
**Pull request by hipstersmoothie**
# What's Changing and Why

determine if an image is using the alpha channel.

closes  #271

## Tasks

- [x] Add tests
- [x] Update Documentation
- [x] Update `jimp.d.ts`
- [x] Add [SemVer](https://semver.org/) Label

### CODE REVIEW

File: packages/jimp/test/filetypes.test.js [hunk 1]
```diff
@@ -135,3 +135,17 @@ describe('FileType', () => {
     image.bitmap.data.should.be.deepEqual(expectedImg.bitmap.data);
   });
 });
+
+describe('hasAlpah', () => {
```
Comment (edi9999) [inline packages/jimp/test/filetypes.test.js]: alpha (typo)

File: packages/core/src/index.js [hunk 2]
```diff
@@ -735,6 +735,27 @@ class Jimp extends EventEmitter {
...
+  hasAlpha() {
+    let hasAlpha = false;
+
+    for (let yIndex = 0; yIndex < this.bitmap.height && !hasAlpha; yIndex++) {
+      for (let xIndex = 0; xIndex < this.bitmap.width && !hasAlpha; xIndex++) {
+        const idx = (this.bitmap.width * yIndex + xIndex) << 2;
+        const alpha = this.bitmap.data[idx + 3];
+
+        if (alpha !== 0xff) {
+          hasAlpha = true;
```
Comment (edi9999) [inline packages/core/src/index.js]: I'm not sure how well v8 and others read the code ahead, but wouldn't it be faster to return early ?

File: README.md [hunk 3]
```diff
@@ -11,6 +11,8 @@ An image processing library for Node written entirely in JavaScript, with zero n
 
 Installation: `npm install --save jimp`
 
+API documnetation can be found in the main [jimp package](./packages/jimp)
```
Comment (edi9999) [inline README.md]: typo documentation

File: packages/core/src/index.js (hunk 2 above)
Comment (hipstersmoothie) [reply packages/core/src/index.js]: the for loops check for hasAlpha to be false so once if find an alpha channel both the loops break. I will go through every pixel if it doesn't find an alpha channel though.

**Comment by hipstersmoothie** [comment]
@edi9999 comments addressed

**Pull request merged**